libraries/AP_BattMonitor/AP_BattMonitor_FuelLevel_Analog.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Code by Charlie Johnson
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+#include "AP_BattMonitor_FuelLevel_Analog.h"
+
+extern const AP_HAL::HAL& hal;
+
+// Constructor
+AP_BattMonitor_FuelLevel_Analog::AP_BattMonitor_FuelLevel_Analog(AP_BattMonitor &mon, AP_BattMonitor::BattMonitor_State &mon_state, AP_BattMonitor_Params &params) :
+    AP_BattMonitor_Analog(mon, mon_state, params)
+{
+    _analog_source = hal.analogin->channel(_curr_pin);
+
+    // create a low pass filter
+    _voltage_filter.set_cutoff_frequency(0.3f);
+}
+
+/*
+  read - read the "voltage" and "current"
+*/
+void AP_BattMonitor_FuelLevel_Analog::read()
+{
+    if (!_analog_source->set_pin(_curr_pin)) {
+        _state.healthy = false;
+        return;
+    }
+    _state.healthy = true;
+
+    // get a dt and a timestamp
+    const uint32_t tnow = AP_HAL::micros();
+    const uint32_t dt_us = tnow - _state.last_time_micros;
+
+    // get votlage from an ADC pin and filter it
+    const float voltage = _analog_source->voltage_average();
+    const float filtered_voltage = _voltage_filter.apply(voltage, float(dt_us) * 1e-3f);
```

Comment:
needs to be 1.0e-6 to get it in seconds

Reply:
a param table for this backend would make sense, and add a filter frequency in Hz as an AP_Float